protoarray/src/main/java/tech/pegasys/teku/protoarray/ProtoArrayStorageChannel.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.protoarray;
+
+import java.util.Optional;
+import tech.pegasys.teku.util.async.SafeFuture;
+
+public interface ProtoArrayStorageChannel {
+
+  void updateProtoArrayOnDisk(ProtoArray protoArray);
```

Comment:
```suggestion
  void onProtoArrayUpdate(ProtoArray protoArray);
```

Reply:
Sending the same mutable instance of `ProtoArray` over the event channel isn't a good idea.  We'll wind up with multiple threads accessing it and what actually gets stored won't be the same as what was there when the call to this method was made.

Reply:
Done.

Reply:
I'm thinking about my options. If I sent a `final ProtoArray` instance we'll still wind up with multiple threads accessing it since it is only a reference. Seems like the best way forward is to convert to bytes on the client-side and send only bytes over the event channel.

Reply:
After offline discussion with @mbaxter, I'm moving on to taking a snapshot of the ProtoArray and sending that over instead.

Reply:
Done.